pkg/keys/sql.go

```diff
@@ -65,6 +65,45 @@ func DecodeTenantPrefixE(key roachpb.Key) ([]byte, roachpb.TenantID, error) {
 	return rem, id, nil
 }
 
+func StripTenantPrefixE(key roachpb.Key) ([]byte, error) {
```

Comment:
[nit] We only added the `E` postfix to DecodeTenantPrefixE to differentiate it from the version that panics. We can probably avoid that here.

[nit] Might want to add a documentation comment.